src/Engine/ProtoCore/Lang/Replication/Replicator.cs

```diff
@@ -214,33 +214,38 @@ public static List<ReplicationInstruction> ReductionToInstructions(List<int> red
                 {
                     //This should generally be a collection, so we need to do a one phase unboxing
                     var targets = reducedParams.Select(r => r[index]).ToList();
-                    foreach (var target in targets)
+                    StackValue target = basicList[index];
```

Comment:
can target be null here, should we check that first?

Reply:
Target cannot be null as we getting its value from basicList and formalParams value is passed into this function based on the input types.

Reply:
👍